product_margin_classification/models/product_margin_classification.py

```diff
@@ -0,0 +1,188 @@
+        string="Total Products With Correct Price",
+        store=True,
+        compute="_compute_template_different_price_qty",
+    )
+
+    template_incorrect_price_qty = fields.Integer(
+        string="Total Products With Incorrect Price",
+        store=True,
+        compute="_compute_template_different_price_qty",
+    )
+
+    template_too_cheap_qty = fields.Integer(
+        string="Total Products Too Cheap",
+        store=True,
+        compute="_compute_template_different_price_qty",
+    )
+
+    template_too_expensive_qty = fields.Integer(
+        string="Total Products Too Expensive",
+        store=True,
+        compute="_compute_template_different_price_qty",
+    )
+
+    price_round = fields.Float(
+        string="Price Rounding",
+        digits=dp.get_precision("Product Price"),
+        default=lambda s: s._default_price_round(),
+        help="Sets the price so that it is a multiple of this value.\n"
+        "Rounding is applied after the margin and before the surcharge.\n"
+        "To have prices that end in 9.99, set rounding 1, surcharge -0.01",
+    )
+
+    price_surcharge = fields.Float(
+        string="Price Surcharge",
+        digits=dp.get_precision("Product Price"),
+        help="Specify the fixed amount to add or substract(if negative) to"
+        " the amount calculated with the discount.",
+    )
+
+    # Default Section
+    @api.model
+    def _default_company_id(self):
+        return self.env.user.company_id.id
+
+    @api.model
+    def _default_price_round(self):
+        decimal_obj = self.env["decimal.precision"]
+        return 10 ** (-decimal_obj.precision_get("Product Price"))
+
+    # constrains Section
+    @api.constrains("markup")
+    def _check_markup(self):
+        for classification in self:
+            if classification.markup == -100:
+                raise UserError(_("-100 is not a valid Markup."))
+
+    @api.multi
+    def _inverse_profit_margin(self):
+        pass
+
+    @api.onchange("profit_margin")
+    def _onchange_profit_margin(self):
+        for classification in self:
+            if classification.profit_margin == 100:
+                raise UserError(_("100 is not a valid Profit Margin."))
+            classification.markup = 100 * (
+                classification.profit_margin
+                / (100 - classification.profit_margin)
+            )
+
+    # Compute Section
+    @api.multi
+    @api.depends("markup")
+    def _compute_profit_margin(self):
+        for classification in self:
+            if classification.markup != -100:
+                classification.profit_margin = (
+                    1 - (1 / (classification.markup / 100 + 1))
+                ) * 100
+
+    @api.multi
+    @api.depends("template_ids.theoretical_difference")
+    def _compute_template_different_price_qty(self):
+        for classification in self:
+            template_vals = classification.template_ids.read(["margin_state"])
```

Comment:
You MUST depends on all fields for stored values. So, template_ids.margin_state.

With this implementation, you do an SQL query at each loop... So, get confident in Odoo ORM and use the good depends.

Reply:
Solved. I set store to False.